lib/versionReplacer.ts

```diff
@@ -47,17 +47,28 @@ export const updateK8sYaml = (
  * @throws {Error} When the old version could not be found in the pubspec.yaml file content.
  */
 export const updatePubspecVersion = (pubspecContent: string, oldVersion: string, newVersion: string): string => {
-  const regex = `version:\\s+(${oldVersion})`;
+  const regex = `version:\\s+(${oldVersion})(?:(?:\\+)(\\d))?`;
 
   const match = pubspecContent.match(regex);
 
   if (match === null) {
     throw new Error("Could not match old version in pubspec.yaml.");
   }
 
-  return pubspecContent.replace(match[0], match[0].replace(match[1], newVersion));
+  let versionString = match[0].replace(match[1], newVersion);
+
+  if (hasBuildNumber(versionString)) {
+    versionString = incrementBuildNumber(versionString, match[2]);
+  }
+
+  return pubspecContent.replace(match[0], versionString);
 };
 
+const hasBuildNumber = (versionString: string) => versionString.includes("+");
+
+const incrementBuildNumber = (versionString: string, oldBuildNumber: string) =>
+  versionString.replace(`+${oldBuildNumber}`, `+${(parseInt(oldBuildNumber) + 1).toString()}`);
```

Comment:
Should maybe catch non parseable build number and throw an error?

Reply:
I see you merged it. Thanks for that. So we don't do anything about this ?

Reply:
Feel free to change this, but I think for now it's ok as it is.